x/sync/network_server.go

```diff
@@ -50,146 +48,70 @@ var (
 	errInvalidEndKey        = errors.New("end key is Nothing but has value")
 	errInvalidBounds        = errors.New("start key is greater than end key")
 	errInvalidRootHash      = fmt.Errorf("root hash must have length %d", hashing.HashLen)
-)
 
-type NetworkServer struct {
-	appSender common.AppSender // Used to respond to peer requests via AppResponse.
-	db        DB
-	log       logging.Logger
-}
+	_ p2p.Handler = (*SyncGetChangeProofHandler)(nil)
+	_ p2p.Handler = (*SyncGetRangeProofHandler)(nil)
+)
 
-func NewNetworkServer(appSender common.AppSender, db DB, log logging.Logger) *NetworkServer {
-	return &NetworkServer{
-		appSender: appSender,
-		db:        db,
-		log:       log,
+func maybeBytesToMaybe(mb *pb.MaybeBytes) maybe.Maybe[[]byte] {
+	if mb != nil && !mb.IsNothing {
```

Comment:
I think `mb.IsNothing` could just be removed right? Because `mb == nil` already implies `Nothing`. Should we remove this field from the type? Or is this needed for some other language?

To clarify - the `mb.IsNothing` check is needed if it's on the type... But I think we could get rid of it entirely.

Reply:
This is code that existed previously but will get rid of the `IsNothing` check

Reply:
I think this is showing up as a diff because of some code that got moved around it but this is previously existing code

Reply:
A future improvement then haha